File: Go/1021 - Notas e moedas.go
package main

import (
	"fmt"
)

func main(){

	var notas, resto, moedas float64
	var n100, n50, n20, n10, n05, n02, n01 float64
	var p50, p25, p10, p05, p01 float64

	fmt.Scan(&notas)

	moedas = (notas-float64(int(notas))) * 100

	n100 = notas/100
	resto = float64(int(notas)%100)

	n50 = resto/50
	resto = float64(int(resto)%50)

	n20 = resto/20
	resto = float64(int(resto)%20)

	n10 = resto/10
	resto = float64(int(resto)%10)

	n05 = resto/05
	resto = float64(int(resto)%05)

	n02 = resto/02
	n01 = float64(int(resto)%02)

	p50 = moedas/50
	resto = float64(int(moedas)%50)

	p25 = resto/25
	resto = float64(int(resto)%25)

	p10 = resto/10
	resto = float64(int(resto)%10)

	p05 = resto/05
	p01 = float64(int(resto)%05)

	fmt.Printf("NOTAS:\n")
	fmt.Printf("%d nota(s) de R$ 100.00\n", int(n100))
	fmt.Printf("%d nota(s) de R$ 50.00\n", int(n50))
	fmt.Printf("%d nota(s) de R$ 20.00\n", int(n20))
	fmt.Printf("%d nota(s) de R$ 10.00\n", int(n10))
	fmt.Printf("%d nota(s) de R$ 5.00\n", int(n05))
	fmt.Printf("%d nota(s) de R$ 2.00\n", int(n02))
	
	fmt.Printf("MOEDAS:\n")
	fmt.Printf("%d moeda(s) de R$ 1.00\n", int(n01))
	fmt.Printf("%d moeda(s) de R$ 0.50\n", int(p50))
	fmt.Printf("%d moeda(s) de R$ 0.25\n", int(p25))
	fmt.Printf("%d moeda(s) de R$ 0.10\n", int(p10))
	fmt.Printf("%d moeda(s) de R$ 0.05\n", int(p05))
	fmt.Printf("%d moeda(s) de R$ 0.01\n", int(p01))
}

File: Lua/1021 - Notas e moedas.lua
function to_int(valor)

	valor = tostring(valor)
	nota = ""
	flag = 0

	for i=1, string.len(valor) do
		
		char = string.sub(valor, i, i)

		if char == "." then
			flag = 1
		elseif flag == 0 then
			nota = nota .. char
		end
	end

	return tonumber(nota)
end

function get_float_part_to_int(valor)

	valor = tostring(valor)
	nota = ""
	flag = 0

	for i=1, string.len(valor) do
		
		char = string.sub(valor, i, i)

		if char == "." then
			flag = 1
		elseif flag == 1 then
			nota = nota .. char
		end
	end

	return tonumber(nota)
end

notas = io.read()
	
n100 = to_int(notas/100)
resto = notas%100

n50 = to_int(resto/50)
resto = resto%50

n20 = to_int(resto/20)
resto = resto%20

n10 = to_int(resto/10)
resto = resto%10

n05 = to_int(resto/5)
resto = resto%5

n02 = to_int(resto/2)
n01 = to_int(resto%2)

moedas = get_float_part_to_int(notas)

p50 = to_int(moedas/50)
resto = moedas%50

p25 = to_int(resto/25)
resto = resto%25

p10 = to_int(resto/10)
resto = resto%10

p05 = to_int(resto/5)
p01 = resto%5

print("NOTAS:")
print(n100 .. " nota(s) de R$ 100.00")
print(n50 .. " nota(s) de R$ 50.00")
print(n20 .. " nota(s) de R$ 20.00")
print(n10 .. " nota(s) de R$ 10.00")
print(n05 .. " nota(s) de R$ 5.00")
print(n02 .. " nota(s) de R$ 2.00")
print("MOEDAS:")
print(n01 .. " moeda(s) de R$ 1.00")
print(p50 .. " moeda(s) de R$ 0.50")

print(p25 .. " moeda(s) de R$ 0.25")
print(p10 .. " moeda(s) de R$ 0.10")
print(p05 .. " moeda(s) de R$ 0.05")
print(p01 .. " moeda(s) de R$ 0.01")

